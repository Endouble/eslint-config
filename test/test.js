const defaults = require('../.');
const react = require('../react');

describe('test basic properties of all configurations', () => {
    it('should match the snapshot', () => {
        expect(defaults).toMatchSnapshot();
        expect(react).toMatchSnapshot();
    });

    it('should be an object', () => {
        expect(defaults).toBeInstanceOf(Object);
        expect(react).toBeInstanceOf(Object);
    });

    it('should have the "rules" property', () => {
        expect(defaults.rules).toBeTruthy();
        expect(react.rules).toBeTruthy();
    });

    it('should extend "airbnb-base"', () => {
        expect(defaults.extends).toBe('airbnb-base');
    });

    it('should extend "airbnb" and defaults', () => {
        expect(react.extends).toEqual(['airbnb', '@endouble.com/eslint-config-endouble']);
    });

    it('should have "react" as a plugin', () => {
        expect(react.plugins).toContain('react');
    });
});
