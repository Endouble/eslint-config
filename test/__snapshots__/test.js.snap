// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test basic properties of all configurations should match the snapshot 1`] = `
Object {
  "extends": "airbnb-base",
  "rules": Object {
    "indent": Array [
      "error",
      4,
      Object {
        "SwitchCase": 1,
      },
    ],
    "max-len": Array [
      "error",
      120,
      2,
      Object {
        "ignoreComments": false,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true,
        "ignoreUrls": true,
      },
    ],
  },
}
`;

exports[`test basic properties of all configurations should match the snapshot 2`] = `
Object {
  "extends": Array [
    "airbnb",
    "@endouble.com/eslint-config-endouble",
  ],
  "parser": "babel-eslint",
  "plugins": Array [
    "react",
  ],
  "rules": Object {
    "import/no-extraneous-dependencies": Array [
      "error",
      Object {
        "devDependencies": true,
      },
    ],
    "import/no-unresolved": Array [
      "error",
      Object {
        "amd": true,
        "commonjs": true,
      },
    ],
    "jsx-a11y/anchor-is-valid": Array [
      "error",
      Object {
        "specialLink": Array [
          "to",
        ],
      },
    ],
    "jsx-a11y/href-no-hash": "off",
    "jsx-a11y/label-has-for": Array [
      "error",
      Object {
        "allowChildren": true,
      },
    ],
    "react/jsx-filename-extension": "off",
    "react/jsx-indent": Array [
      "error",
      4,
    ],
    "react/jsx-indent-props": Array [
      "error",
      4,
    ],
  },
}
`;
